app/Services/Locations/InternalRentalBillingService.php

```
@@ -0,0 +1,129 @@
+                $existing->update($attributes);
+            }
+
+            return $existing;
+        }
+
+        try {
+            return $fixedAsset->internalRentalInvoices()->create($attributes + ['billing_key' => $billingKey]);
+        } catch (QueryException) {
+            // Course concurrente : la contrainte unique `billing_key` a déjà été
+            // satisfaite, on renvoie la facture insérée par l'autre requête.
+            return $fixedAsset->internalRentalInvoices()
```

PHPStan (2.2.7)

[error] Method App\Services\Locations\InternalRentalBillingService::generateForAsset() should return App\Models\Locations\InternalRentalInvoice|null but returns Illuminate\Database\Eloquent\Model|null.


(return.type)